WooCommerce/Classes/POS/Adaptors/View Adaptors/POSCouponCreationSheetAdaptor.swift

```diff
@@ -0,0 +1,74 @@
+import SwiftUI
+import struct Yosemite.Coupon
+
+/// Provide access to AddEditCouponView and ViewModel for POS without making it as an explicit type dependency
+/// AddEditCouponView cannot be easily moved and reused in a shared module due to multiple dependencies
+/// This is used as a workaround to enable POS modularization without requiring a larger refactoring effort
+///
+
+struct POSCouponCreationViewAdaptor: View {
+    @StateObject private var viewModel: AddEditCouponViewModel
+    @Binding var showTypeSelection: Bool
+    let dismissHandler: () -> Void
+    let onDisappear: () -> Void
```

Comment:
nit: these could be private.
```suggestion
    @Binding private var showTypeSelection: Bool
    private let dismissHandler: () -> Void
    private let onDisappear: () -> Void
```